offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -273,23 +273,59 @@ class ProductsDbConnector implements ArchiveProductDataSource, CreateProductData
    * @throws ProductExistsException if the product already exists in the data source
    */
   @Override
-  void store(Product product) throws DatabaseQueryException, ProductExistsException {
+  ProductId store(Product product) throws DatabaseQueryException, ProductExistsException {
     Connection connection = provider.connect()
 
+    ProductId id = createProductId(product)
+
```

Comment:
Maybe we can discuss that in a meeting? I also thought about what you suggested. I think it makes sense to only create the ID with the DB. You cannot know the latest version in the view since we also have archived products and therefore "hidden" products with ids.

Reply:
let's talk about it. I do not fully agree with your statement that the view has no access to the running numbers